src/pandapipes/pf/derivative_calculation.py

```diff
@@ -94,29 +110,121 @@ def calculate_derivatives_thermal(net, branch_pit, node_pit, _):
     to_nodes = get_to_nodes_corrected(branch_pit)
     t_init_i = node_pit[from_nodes, TINIT_NODE]
     t_init_i1 = branch_pit[:, TOUTINIT]
-    t_init_n = node_pit[to_nodes, TINIT_NODE]
-    cp_n = fluid.get_heat_capacity(t_init_n)
+    t_init_n = node_pit[:, TINIT_NODE]
+    cp_i = fluid.get_heat_capacity(t_init_i)
     cp_i1 = fluid.get_heat_capacity(t_init_i1)
+    cp_n = fluid.get_heat_capacity(t_init_n)
     t_amb = branch_pit[:, TEXT]
     length = branch_pit[:, LENGTH]
     alpha = branch_pit[:, ALPHA] * np.pi * branch_pit[:, D]
     tl = branch_pit[:, TL]
```

Comment:
For Understanding:

Temperature lift TL comes from where and what does it represent in the equation?

Reply:
I am  not sure if it is used in any component model, but it is just an equivalent to the pressure lift of hydraulics calculation.